TLM/TLM/State/ConfigData/DebugSettings.cs

```diff
@@ -135,9 +140,46 @@ public enum DebugSwitch {
         // Added a new flag? Bump LATEST_VERSION in GlobalConfig!
     }
 
-    static class DebugSwitchExtensions {
+    public class DebugSwitchValue : IXmlSerializable {
+        public bool Value = false;
+        public string Description = string.Empty;
+
+        [UsedImplicitly] //required for XmlSerializer
+        public DebugSwitchValue() { }
+
+        public DebugSwitchValue(bool value, string description) {
+            Value = value;
+            Description = description;
+        }
+        public XmlSchema GetSchema() {
+            return null;
+        }
+
+        public void ReadXml(XmlReader reader) {
+            while (reader.Read()) {
+                switch (reader.NodeType) {
+                    case XmlNodeType.Text:
+                        bool.TryParse(reader.Value, out Value);
+                        break;
+                    case XmlNodeType.Comment:
+                        Description = reader.Value;
+                        break;
+                    case XmlNodeType.EndElement:
+                        return;
+                }
+            }
+        }
+
+        public void WriteXml(XmlWriter writer) {
+            writer.WriteValue(Value);
+            writer.WriteComment(Description);
+            writer.WriteString(string.Empty); //adds line break
```

Comment:
I don't understand the line break part. add line break where?

Reply:
oh, right. I've had to add it before because some weird issues with formatting then I implemented read/write and problem disappeared. Not needed anymore 👍